src/main/java/frc/robot/commands/swervedrive/drivebase/TeleopDrive.java

```diff
@@ -103,6 +110,13 @@ else if(isBackSpeaker)
       angVelocity *= slowFactor;
     }
 
+    if(isHeadingLock)
+    {
+       PID.setSetpoint(UtilMath.SourceIntakeHeading(swerve.getPose()));
+       double result =  PID.calculate(currTheta)/4;
+       angVelocity += result;
```

Comment:
Why is this a plus-equals operation? I think you intended it to be like the other ones - set the desired angular velocity based on the PID calculation